tests/datasets/configuration/test_generic_metrics.py

```diff
@@ -1,6 +1,66 @@
+import pytest
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
 from yaml import safe_load
 
+from snuba.clickhouse.columns import AggregateFunction, Column, Float
+from snuba.datasets.configuration.json_schema import (
+    readable_storage_schema,
+    writable_storage_schema,
+)
+from snuba.datasets.configuration.utils import parse_columns
+from snuba.datasets.storages.generic_metrics import (
+    aggregate_common_columns,
+    bucket_columns,
+    common_columns,
+)
 
-def test_file_loads() -> None:
-    file = open("./snuba/datasets/configuration/generic_metrics/dataset.yaml")
-    safe_load(file)
+CONF_PATH = "./snuba/datasets/configuration/generic_metrics/storages"
+
+DISTRIBUTIONS_STORAGE_COLUMN_SET = [
+    *common_columns,
+    *aggregate_common_columns,
+    Column(
+        "percentiles",
+        AggregateFunction("quantiles(0.5, 0.75, 0.9, 0.95, 0.99)", [Float(64)]),
+    ),
+    Column("min", AggregateFunction("min", [Float(64)])),
+    Column("max", AggregateFunction("max", [Float(64)])),
+    Column("avg", AggregateFunction("avg", [Float(64)])),
+    Column("sum", AggregateFunction("sum", [Float(64)])),
+    Column("count", AggregateFunction("count", [Float(64)])),
+    Column(
+        "histogram_buckets",
+        AggregateFunction("histogram(250)", [Float(64)]),
+    ),
+]
+
+
+def test_distributions_storage() -> None:
+    file = open(f"{CONF_PATH}/distributions_storage.yaml")
+    config = safe_load(file)
+    validate(config, readable_storage_schema)
```

Comment:
I think `parse YAML and validate config` should be encapsulated in a single `load_config` method/class somewhere and we should strongly discourage any sideload that doesn't also do enforcement

Reply:
Done, I've created a function that loads a config given a `StorageKey`.